pkg/network/tracer/connection/failure/matching.go

```diff
@@ -110,17 +121,26 @@ func (fc *FailedConns) MatchFailedConn(conn *network.ConnectionStats) {
 	util.ConnStatsToTuple(conn, fc.failureTuple)
 
 	fc.RLock()
-	defer fc.RUnlock()
+	foundMatch := false
+	var failedConn *FailedConnStats
 
 	if failedConn, ok := fc.FailedConnMap[*fc.failureTuple]; ok {
 		// found matching failed connection
+		foundMatch = true
 		conn.TCPFailures = make(map[uint32]uint32)
```

Comment:
It may be better to transfer ownership of the map to the connection, setting `failedConn.CountByErrCode = nil`, and then letting `failedConn` getting GCed. Definitely worth profiling, but if you need to allocate a new map anyways, I'm not sure a `sync.Pool` is helpful.

Reply:
Bryce and I talked offline but we decided to go with this to avoid the extra map allocation